File: library-app/src/main/kotlin/com/group/libraryapp/calculator/domain/book/BookRepository.java
package com.group.libraryapp.calculator.domain.book;

import com.group.libraryapp.domain.book.Book;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface BookRepository extends JpaRepository<com.group.libraryapp.domain.book.Book, Long> {

  Optional<Book> findByName(String bookName);

}

File: library-app/src/main/kotlin/com/group/libraryapp/calculator/domain/book/BookRepository.kt
package com.group.libraryapp.calculator.domain.book

import com.group.libraryapp.calculator.domain.book.Book
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface BookRepository : JpaRepository<Book, Long> {


    //java
    //Optional<Book> findByName(String bookName);
    fun findByName(bookName: String): Optional<Book>

}